dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/config/validation/ConfigurationValidator.java

```diff
@@ -58,6 +61,70 @@ public ValidationResult validate(IResource resource, int kind, ValidationState s
             Trace.logError("Error removing markers", e);
         }
 
+        // For defined runtimes, do not validate any server config files in a templates folder inside the runtime folder
+        IPath fileLocation = file.getLocation(); // Full system path of the file
+        IPath fileFullPath = file.getFullPath(); // Used to ensure that templates is a subfolder of the runtime
+        WebSphereRuntime[] webSphereRuntimes = WebSphereUtil.getWebSphereRuntimes();
+        int length = webSphereRuntimes.length;
+        for (int i = 0; i < length; i++) {
+            WebSphereRuntime runtime = webSphereRuntimes[i];
+            IPath runtimeLocation = runtime.getRuntimeLocation();
+            if (runtimeLocation.isPrefixOf(fileLocation) && fileFullPath.toString().contains("templates")) { //$NON-NLS-1$
+                return EMPTY_RESULT;
+            }
+        }
+
+        // Workaround to not validate server.xml files in templates folder in build plugins
+        // eg. target/liberty/wlp/templates
+        // eg. build/wlp/templates/
+        // Best guess to find a templates folder in a 'ghost' runtime
+        // This is the simplest check: if the file is in a path that contains wlp/templates
+        if (fileFullPath.toString().contains("wlp/templates")) { //$NON-NLS-1$
+            return EMPTY_RESULT;
+        }
+        // Do another check since the folder names can be configured.
+        // The templates folder is a sibling of the bin folder.  So check if the bin folder exists
+        // and it has a server script
+        // Two conditions:
+        // 1) it must be in some templates folder
+        boolean hasTemplates = fileFullPath.toString().contains("templates"); //$NON-NLS-1$
+        // 2) the templates folder is a sibling of the Liberty runtime bin folder which must contain a server script.  This means
+        //    other templates folders will still be validated.
+        if (hasTemplates) {
+            IContainer parent = file.getParent(); // get parent containers until we reach templates
```

Comment:
Would it be better to use the existing static isValidLocation method in WebSphereRuntime to check if this is a WebSphere runtime since this is what is used elsewhere?

Reply:
Done